apollo-router/src/http_ext.rs

```diff
@@ -216,27 +216,22 @@ pub(crate) fn header_map(
     Ok(http)
 }
 
-/// Ignores `http::Extensions`
+// In an earlier version of the http crate, `Request` didn't implement `Clone`, so we
+// implemented partial support (excluding Extensions) for `Clone`. `Request` does now implement
+// `Clone`, so this function does now clone Requests fully.
+//  - https://github.com/hyperium/http/pull/634
+//  - https://github.com/hyperium/http/releases/tag/v1.0.0
 pub(crate) fn clone_http_request<B: Clone>(request: &http::Request<B>) -> http::Request<B> {
-    let mut new = http::Request::builder()
-        .method(request.method().clone())
-        .uri(request.uri().clone())
-        .version(request.version())
-        .body(request.body().clone())
-        .unwrap();
-    *new.headers_mut() = request.headers().clone();
-    new
+    request.clone()
 }
 
-/// Ignores `http::Extensions`
+// In an earlier version of the http crate, `Response` didn't implement `Clone`, so we
+// implemented partial support (excluding Extensions) for `Clone`. `Response` does now implement
+// `Clone`, so this function does now clone Responses fully.
+//  - https://github.com/hyperium/http/pull/634
+//  - https://github.com/hyperium/http/releases/tag/v1.0.0
 pub(crate) fn clone_http_response<B: Clone>(response: &http::Response<B>) -> http::Response<B> {
```

Comment:
I think we should remove these functions and replace the callsites. Primarily because the comment in this callsite is now incorrect https://github.com/apollographql/router/blob/678dee7e1a85d4520d84e90ef0913b892a55ee65/apollo-router/src/services/router/service.rs#L709-L710

Reply:
Oh, I didn't see you updated the comment, then it doesn't matter :)